src/pybind/mgr/dashboard/services/rgw_client.py

```diff
@@ -645,6 +645,45 @@ def set_bucket_versioning(self, bucket_name, versioning_state, mfa_delete,
                                      http_status_code=error.status_code,
                                      component='rgw')
 
+    @RestClient.api_get('/{bucket_name}?encryption')
+    def get_bucket_encryptyion(self, bucket_name, request=None):
+
+        # pylint: disable=unused-argument
+        result = request()
+        if 'SSEAlgorithm' not in result:
+            result['SSEAlgorithm'] = 'No Encryption'
+        return result
+
+    @RestClient.api_put('/{bucket_name}?encryption')
+    def set_bucket_encryption(self, bucket_name, keyId, bucketKeyEnabled, sseAlgorithm, request=None):
+
+        # pylint: disable=unused-argument
+        encryption_configuration = ET.Element('ServerSideEncryptionConfiguration')
+        rule_element = ET.SubElement(encryption_configuration, 'Rule')
+
+        default_encryption_element = ET.SubElement(rule_element,
+                                                  'ApplyServerSideEncryptionByDefault')
+
+        bucket_key_enabled_element = ET.SubElement(rule_element,
+                                                  'BucketKeyEnabled')
+        bucket_key_enabled_element.text = bucketKeyEnabled
+
+        sse_algo_element = ET.SubElement(default_encryption_element,
+                                                  'SSEAlgorithm')
+        sse_algo_element.text = sseAlgorithm
+
+        if sseAlgorithm == 'aws:kms':
+            kms_master_key_element = ET.SubElement(default_encryption_element,
+                                                  'KMSMasterKeyID')
+            kms_master_key_element.text = keyId
+
+        data = ET.tostring(encryption_configuration, encoding='unicode')
+
+        try:
+            _ = request(data=data)  # type: ignore
+        except RequestException as e:
+            raise DashboardException(msg=str(e), component='rgw')
```

Comment:
We should use a decorator to wrap these exceptions (from `services/exceptions.py`) and avoid the boilerplate try-except code.